### PR TITLE
Sto tests

### PR DESCRIPTION
**Pull request by victorwiebe**

### CODE REVIEW

File: src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts [hunk 1]
```diff
@@ -272,6 +273,188 @@ describe('CappedSTOWrapper', () => {
...
+        safetyFactor: 10,
+      };
+      const expectedResult = getMockedPolyResponse();
+      // Mocked method
+      const mockedMethod = mock(MockedSendMethod);
+      // Stub the method
+      when(mockedContract.buyTokens).thenReturn(instance(mockedMethod));
+      // Stub the request
+      when(
+        mockedMethod.sendTransactionAsync(mockedParams.beneficiary, txPayableData, mockedParams.safetyFactor),
```
Comment (VictorVicente) [inline src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts]: Problem is the same we discussed recently. `txPayable` data is a different instance here and in the real call. You should use `objectContaining` instead.

File: src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts [hunk 2]
```diff
@@ -272,6 +273,188 @@ describe('CappedSTOWrapper', () => {
...
+      // Stub the method
+      when(mockedContract.endTime).thenReturn(instance(mockedEndTimeMethod));
+      // Stub the request
+      when(mockedEndTimeMethod.callAsync()).thenResolve(expectedEndTimeResult);
+
+      // ****** Real method setup and call *****
+      const txPayableData = {
+        ...{},
+        from: expectedOwnerResult,
+        value: new BigNumber(1),
```
Comment (VictorVicente) [inline src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts]: You should move `txPayableData` definition after the `mockedParams` definition and value should be `mockedParams.value`.
Lesser hardcoded value, easier to read and to change tests in the future.

Comment (victorwiebe) [reply src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts]: Done

File: src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts (hunk 1 above)
Comment (victorwiebe) [reply src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts]: Okay used objectContaining(txPayable) seems to work

File: src/contract_wrappers/registries/security_token_registry_wrapper.ts [hunk 3]
```diff
@@ -810,7 +810,7 @@ export default class SecurityTokenRegistryWrapper extends ContractWrapper {
     if (registrationDate.getTime() === new Date(0).getTime()) {
       return true;
     }
-    if (!isDeployed && expiryDate.getTime() > Date.now()) {
+    if (!isDeployed && expiryDate.getTime() < Date.now()) {
```
Comment (VictorVicente) [inline src/contract_wrappers/registries/security_token_registry_wrapper.ts]: I think this is wrong. If `expiryDate.getTime()` is less than current time, the ticker is reserved, so it is not available.

File: src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+      // Result expectation
+      expect(result).toBe(expectedResult);
+      // Verifications
+      verify(mockedContract.allowBeneficialInvestments).once();
+      verify(mockedMethod.callAsync()).once();
+    });
+  });
+
+  describe('Investors', () => {
+    test('should get bigNumber for given investor', async () => {
```
[the remark this answers is not in the document]
Comment (victorwiebe) [reply src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts]: Not sure what happened there. Committing investor test now.

File: src/contract_wrappers/registries/security_token_registry_wrapper.ts (hunk 3 above)
Comment (victorwiebe) [reply src/contract_wrappers/registries/security_token_registry_wrapper.ts]: Reversed this change

File: src/contract_wrappers/modules/sto/__tests__/sto_wrapper.test.ts [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+  });
+
+  afterEach(() => {
+    reset(mockedWrapper);
+    reset(mockedContract);
+    reset(mockedSecurityTokenContract);
+  });
+
+  describe('Types', () => {
+    test('should extend ContractWrapper', async () => {
```
[the remark this answers is not in the document]
Comment (victorwiebe) [reply src/contract_wrappers/modules/sto/__tests__/sto_wrapper.test.ts]: Fixed in both branches

File: src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+      verify(
+        mockedMethod.sendTransactionAsync(
+          mockedParams.allowBeneficialInvestments,
+          mockedParams.txData,
+          mockedParams.safetyFactor,
+        ),
+      ).once();
+      verify(mockedSecurityTokenOwnerMethod.callAsync()).once();
+      verify(mockedSecurityTokenContract.owner).once();
+      verify(mockedAllowBeneficialInvestmentMethod.callAsync()).once();
```
[the remark this answers is not in the document]
Comment (victorwiebe) [reply src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts]: Going to go back through all my tests to look for this, I wasn't very proactive on this in earlier tests.

**Comment by VictorVicente** [comment]
Hey @victorwiebe, one test is failing. Could you review it?

**Comment by victorwiebe** [comment]
@victorvicente can you please pull locally and check it? The issue is with buyTokensView. The target code runs fine, but when it gets to the verifys statements it fails on isOpen and several others. I have spent a lot of time on this test and can’t figure it out- all the code appears to be setup correctly to me already.

**Pull request merged**